pkg/csi-common/utils.go

```diff
@@ -63,26 +80,28 @@ func NewNodeServiceCapability(cap csi.NodeServiceCapability_RPC_Type) *csi.NodeS
 	}
 }
 
-func getLogLevel(method string) int32 {
-	if method == "/csi.v1.Identity/Probe" ||
-		method == "/csi.v1.Node/NodeGetCapabilities" ||
-		method == "/csi.v1.Node/NodeGetVolumeStats" ||
-		method == "/csi.v1.Controller/ListVolumes" {
-		return 6
+func getLogLevel(method string) logLevels {
+	if levels, ok := lowPriApis[method]; ok {
+		return levels
 	}
-	return 2
+
+	return defaultLogLevels
 }
 
 func logGRPC(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
-	level := klog.Level(getLogLevel(info.FullMethod))
-	klog.V(level).Infof("GRPC call: %s", info.FullMethod)
-	klog.V(level).Infof("GRPC request: %s", protosanitizer.StripSecrets(req))
+	levels := getLogLevel(info.FullMethod)
+
+	klog.V(levels.verbose).Infof("GRPC call: %s, request: %s", info.FullMethod, protosanitizer.StripSecrets(req))
```

Comment:
I am ok to combine `GRPC call` and `GRPC request` logging as one, but we still need same level logging for `GRPC call: xxx`, so change this from `levels.verbose` to `levels.normal`

Reply:
@andyzhangx Why would we need to emit the `GRPC call` trace at `normal` level when we can deteemine the start of call from the `GRPC (succeeded|failed)` trace's timestamp and latency?

Reply:
after another look at the logs, I think original logging is more clear since we could get all operation details between `GRPC call` and `GRPC response`, it may better to enrich `GRPC response` log a little:

 - original logs
```
I0303 09:08:02.272503       1 utils.go:77] GRPC call: /csi.v1.Controller/ControllerPublishVolume
I0303 09:08:02.272525       1 utils.go:78] GRPC request: {"node_id":"k8s-agentpool1-31426467-vmss000000","volume_capability":{"AccessType":{"Mount":{}},"access_mode":{"mode":7}},"volume_context":{"csi.storage.k8s.io/pv/name":"pvc-4841c5f3-ce6c-4684-ae79-92d7372a9e87","csi.storage.k8s.io/pvc/name":"pvc-azuredisk","csi.storage.k8s.io/pvc/namespace":"default","requestedsizegib":"10","skuName":"StandardSSD_LRS","storage.kubernetes.io/csiProvisionerIdentity":"1677834472830-8081-test.csi.azure.com"},"volume_id":"/subscriptions/0e46bd28-a80f-4d3a-8200-d9eb8d80cb2e/resourceGroups/kubetest-a21ylw09/providers/Microsoft.Compute/disks/pvc-4841c5f3-ce6c-4684-ae79-92d7372a9e87"}
...
I0303 09:08:02.460544       1 controllerserver.go:383] GetDiskLun returned: cannot find Lun for disk pvc-4841c5f3-ce6c-4684-ae79-92d7372a9e87. Initiating attaching volume /subscriptions/0e46bd28-a80f-4d3a-8200-d9eb8d80cb2e/resourceGroups/kubetest-a21ylw09/providers/Microsoft.Compute/disks/pvc-4841c5f3-ce6c-4684-ae79-92d7372a9e87 to node k8s-agentpool1-31426467-vmss000000 (vmState Succeeded).
I0303 09:08:02.460572       1 controllerserver.go:408] Trying to attach volume /subscriptions/0e46bd28-a80f-4d3a-8200-d9eb8d80cb2e/resourceGroups/kubetest-a21ylw09/providers/Microsoft.Compute/disks/pvc-4841c5f3-ce6c-4684-ae79-92d7372a9e87 to node k8s-agentpool1-31426467-vmss000000
I0303 09:08:02.460613       1 azure_controller_common.go:255] Trying to attach volume /subscriptions/0e46bd28-a80f-4d3a-8200-d9eb8d80cb2e/resourceGroups/kubetest-a21ylw09/providers/Microsoft.Compute/disks/pvc-4841c5f3-ce6c-4684-ae79-92d7372a9e87 lun 0 to node k8s-agentpool1-31426467-vmss000000, diskMap: map[/subscriptions/0e46bd28-a80f-4d3a-8200-d9eb8d80cb2e/resourcegroups/kubetest-a21ylw09/providers/microsoft.compute/disks/pvc-4841c5f3-ce6c-4684-ae79-92d7372a9e87:%!s(*provider.AttachDiskOptions=&{ReadOnly pvc-4841c5f3-ce6c-4684-ae79-92d7372a9e87  false 0})]
I0303 09:08:02.460642       1 azure_controller_vmss.go:110] azureDisk - update(kubetest-a21ylw09): vm(k8s-agentpool1-31426467-vmss000000) - attach disk list(map[/subscriptions/0e46bd28-a80f-4d3a-8200-d9eb8d80cb2e/resourcegroups/kubetest-a21ylw09/providers/microsoft.compute/disks/pvc-4841c5f3-ce6c-4684-ae79-92d7372a9e87:%!s(*provider.AttachDiskOptions=&{ReadOnly pvc-4841c5f3-ce6c-4684-ae79-92d7372a9e87  false 0})])
...
I0303 09:08:13.752310       1 azure_metrics.go:115] "Observed Request Latency" latency_seconds=11.431556177 request="azuredisk_csi_driver_controller_publish_volume" resource_group="kubetest-a21ylw09" subscription_id="0e46bd28-a80f-4d3a-8200-d9eb8d80cb2e" source="test.csi.azure.com" volumeid="/subscriptions/0e46bd28-a80f-4d3a-8200-d9eb8d80cb2e/resourceGroups/kubetest-a21ylw09/providers/Microsoft.Compute/disks/pvc-4841c5f3-ce6c-4684-ae79-92d7372a9e87" node="k8s-agentpool1-31426467-vmss000000" result_code="succeeded"
I0303 09:08:13.752324       1 utils.go:84] GRPC response: {"publish_context":{"LUN":"0"}}
```